src/ring_theory/dedekind_domain/adic_valuation.lean

```diff
@@ -222,4 +240,119 @@ begin
+
+/-- The `v`-adic valuation of `r ∈ R` is less than 1 if and only if `v` divides the ideal `(r)`. -/
+lemma valuation_lt_one_iff_dvd (r : R) :
+  v.valuation (algebra_map R K r) < 1 ↔ v.as_ideal ∣ ideal.span {r} :=
+by { rw valuation_of_algebra_map, exact v.int_valuation_lt_one_iff_dvd r }
+
+variable (K)
+/-- There exists `π ∈ K` with `v`-adic valuation `multiplicative.of_add (-1)`. -/
+lemma valuation_exists_uniformizer :
+  ∃ (π : K), v.valuation π = multiplicative.of_add (-1 : ℤ) :=
+begin
+  obtain ⟨r, hr⟩ := v.int_valuation_exists_uniformizer,
+  use algebra_map R K r,
+  rw [valuation_def, valuation.extend_to_localization_apply_map_apply],
+  exact hr,
+end
+
+/-- Uniformizers are nonzero. -/
+lemma valuation_uniformizer_ne_zero :
+  (classical.some (v.valuation_exists_uniformizer K)) ≠ 0 :=
+begin
+  have hu := (classical.some_spec (v.valuation_exists_uniformizer K)),
+  exact (valuation.ne_zero_iff _).mp (ne_of_eq_of_ne hu with_zero.coe_ne_zero),
+end
+
+/-! ### Completions with respect to adic valuations
+Given a Dedekind domain `R` with field of fractions `K` and a maximal ideal `v` of `R`, we define
+the completion of `K` with respect to its `v`-adic valuation, denoted `v.adic_completion`, and its
+ring of integers, denoted `v.adic_completion_integers`. -/
+
+variable {K}
+
+/-- `K` as a valued field with the `v`-adic valuation. -/
+def adic_valued : valued K (with_zero (multiplicative ℤ)) := valued.mk' v.valuation
+
+lemma adic_valued_def {x : K} : (v.adic_valued.v : _) x = v.valuation x := rfl
+
+/-- The uniform space structure on `K` corresponding to the `v`-adic valuation.
+This cannot be made an instance since it depends on the choice of `v`. -/
+def adic_valued_uniform_space : uniform_space K := v.adic_valued.to_uniform_space
+
+lemma adic_valued_uniform_add_group : @uniform_add_group K v.adic_valued_uniform_space _ :=
+v.adic_valued.to_uniform_add_group
+
+/-- The topological space structure on `K` corresponding to the `v`-adic valuation.
+  This cannot be made an instance since it depends on the choice of `v`. -/
+def adic_valued_topological_space : topological_space K :=
+v.adic_valued_uniform_space.to_topological_space
+
+lemma adic_valued_completable_top_field : @completable_top_field K _ v.adic_valued_uniform_space :=
+@valued.completable K _ _ _ (adic_valued v)
+
+lemma adic_valued_separated_space : @separated_space K v.adic_valued_uniform_space :=
+@valued_ring.separated K _ _ _ (adic_valued v)
+
+variables (K)
+
+/-- The completion of `K` with respect to its `v`-adic valuation. -/
+def adic_completion := @uniform_space.completion K v.adic_valued_uniform_space
+
+instance : field (v.adic_completion K) :=
+@field_completion K _ v.adic_valued_uniform_space _ _ v.adic_valued_uniform_add_group
+
+instance : inhabited (v.adic_completion K) := ⟨0⟩
+
+variables {K}
+instance valued_adic_completion : valued (v.adic_completion K) (with_zero (multiplicative ℤ)) :=
+valued.mk' (@valued.extension_valuation K _ _ _ v.adic_valued)
```

Comment:
This is the exact situation in which you must use `valued.valued_completion` instead of `valued.mk'`:
```suggestion
@valued.valued_completion _ _ _ _ v.adic_valued
```

Reply:
Thanks!